docs/walkthrough/walkthrough-istio.md

```diff
@@ -0,0 +1,384 @@
+- network policies allow to secure pod-to-pod or service-to-service communication at the network and application layers
+
+Some observality benefits:
+- understand how service performance impacts things upstream and downstream
+- help detect and fix issues quickly and efficiently
+
+Some resiliency benefits:
+- automatically retry failed service calls
+- mirror traffic to send a copy of live traffic to a mirrored service
+- add circuit breakers directly on the wire
+- inject network faults to test applications behaviour
+
+The list is by no means exhaustive and service meshes bring a lot more benefits than just those enumerated above.
+
+Visit [istio's webpage](https://istio.io/latest/docs/concepts/what-is-istio/) or [istio's blog](https://istio.io/latest/blog) for more info.
+
+## Creating a local Kubernetes cluster
+
+The first thing to do is to create a Kubernetes cluster.
+
+Kind is an easy solution to run a local cluster, all it needs is to have `docker` installed.
+
+Create a cluster by running the following command:
+
+```bash
+kind create cluster --name walkthrough --config - <<EOF
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  extraPortMappings:
+  - containerPort: 30080
+    hostPort: 80
+    protocol: TCP
+  - containerPort: 30443
+    hostPort: 443
+    protocol: TCP
+EOF
+```
+
+This will create a cluster with ports 80 and 443 exposed on the host computer. This is needed to communicate with the istio ingress gateway.
+
+Once the cluster is started, you can check that the context was added in your kubeconfig file by running:
+
+```bash
+kubectl config get-contexts
+```
+
+The output should look something like this:
+
+```bash
+CURRENT   NAME                                                 CLUSTER                          AUTHINFO                               NAMESPACE
+*         kind-walkthrough                                     kind-walkthrough                 kind-walkthrough                       
+```
+
+The `kind-walkthrough` is the context for the cluster you just created and the `*` at the beginning means that it is the currently active context.
+
+You can verify that the ports were correctly mapped by `docker` by running `docker ps`. The output should show something like this:
+
+```bash
+CONTAINER ID        IMAGE                  COMMAND                  CREATED              STATUS              PORTS                                                                      NAMES
+084e58aa36fe        kindest/node:v1.17.0   "/usr/local/bin/entr…"   About a minute ago   Up 56 seconds       127.0.0.1:32778->6443/tcp, 0.0.0.0:80->30080/tcp, 0.0.0.0:443->30443/tcp   walkthrough-control-plane
+```
+
+## Installing the istio operator
+
+Now that the cluster is up and running, you will need to install istio.
+
+Instead of installing istio directly in the cluster, you will install the `istio-operator` then the operator will take care of installing the istio service mesh inside the cluster.
```

Comment:
```suggestion
Instead of installing Istio directly in the cluster, you will install the `istio-operator` then the operator will take care of installing the Istio service mesh inside the cluster.
```